### PR TITLE
`aes.c` and `aes_api.c`

### PR DESCRIPTION
**Pull request by Thar0**
And imports libcrypto debug info for `aes_imp.c`

### CODE REVIEW

File: include/libcrypto/aes.h [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+    /* 0x08 */ u32 rk[60];
+    /* 0xF8 */ u32 ek[60];
+} AesKeyInstance; // size = 0x1E8
+
+typedef struct {
+    /* 0x0 */ unsigned int mode;
+    /* 0x4 */ unsigned char IV[16];
+} AesCipherInstance; // size = 0x14
+
+void* memcpy(void* dst, void* src, u32 num);
```
Comment (AngheloAlf) [inline include/libcrypto/aes.h]: There's no `libc/string.h`?

File: src/libcrypto/aes_api.c [hunk 2]
```diff
@@ -1,9 +1,125 @@
 #include "include_asm.h"
+#include "ultratypes.h"
+#include "libcrypto/aes.h"
 
-INCLUDE_ASM("asm/non_matchings/libcrypto/aes_api", aesMakeKey);
+int aesMakeKey(AesKeyInstance* key, u8 direction, int keyLen, u8* keyMaterial) {
+    if (key == NULL) {
+        return -3;
+    }
+    if (direction >= 2U) {
```
Comment (ethteck) [inline src/libcrypto/aes_api.c]: Do we need the U here on the 2? or is that convention here and I just am unaware

File: src/libcrypto/aes_api.c [hunk 3]
```diff
@@ -1,9 +1,125 @@
...
+    } else {
+        key->Nr = rijndaelKeySetupDec(key->rk, keyMaterial, keyLen);
+    }
+    rijndaelKeySetupEnc(key->ek, keyMaterial, keyLen);
+    return 1;
+}
 
-INCLUDE_ASM("asm/non_matchings/libcrypto/aes_api", aesBlockDecrypt);
+int aesCipherInit(AesCipherInstance* cipher, u8 mode, u8* IV) {
+    if (mode - 1 >= 2U) {
```
Comment (ethteck) [inline src/libcrypto/aes_api.c:28]: here2

File: include/libcrypto/aes.h (hunk 1 above)
Comment (pixel-stuck) [reply include/libcrypto/aes.h]: They implemented their own libc functions, it seems. We need to build with -fno-builtin

File: src/libcrypto/aes_api.c (hunk 3 above)
Comment (Thar0) [reply src/libcrypto/aes_api.c:28]: It was needed here but not in the other place

**Review by ethteck — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**